### ISSUE TITLE
Use of monitor_models on a pipeline component overwrites all earlier output models on the pipeline task

### ISSUE THREAD

**Issue by tonyd** [allegroai/clearml#757]
Thank you for helping us making ClearML better!

## Describe the bug
Use of the monitor_models param on a pipeline component to copy a model up to the pipeline task causes any output models already on the pipeline task from an earlier step to be discarded, even if there is no name conflict.

## To reproduce
```
from clearml.automation.controller import PipelineDecorator

@PipelineDecorator.component(monitor_models=['model1'])
def step1():
    from clearml import Task, OutputModel

    model = OutputModel(name="model1")
    with open('model1.xxx', 'w') as f: pass
    model.update_weights(weights_filename='model1.xxx')

@PipelineDecorator.component(monitor_models=['model2'])
def step2():
    from clearml import Task, OutputModel

    model = OutputModel(name="model2")
    with open('model2.xxx', 'w') as f: pass
    model.update_weights(weights_filename='model2.xxx')

@PipelineDecorator.pipeline(name='test pipeline monitor models', project='tests', version='1')
def pipeline():
    from clearml import Task

    print(step1())

    print("after step1:")
    PipelineDecorator.get_current_pipeline().update_execution_plot()
    Task.current_task().reload()
    for m in Task.current_task().models.output:
        print(m.name)

    print(step2())

    print("after step2:")
    PipelineDecorator.get_current_pipeline().update_execution_plot()
    Task.current_task().reload()
    for m in Task.current_task().models.output:
        print(m.name)

if __name__ == '__main__':
    PipelineDecorator.run_locally()
    pipeline()
```

## Expected behaviour
```
ClearML Task: created new task id=8f1f99dce1c34821932fdcbf40670bf5
ClearML results page: https://app.clear.ml/projects/323c182596f34d73934d0b7905eeb81d/experiments/8f1f99dce1c34821932fdcbf40670bf5/output/log
ClearML pipeline page: https://app.clear.ml/pipelines/323c182596f34d73934d0b7905eeb81d/experiments/8f1f99dce1c34821932fdcbf40670bf5
Launching step [step1]
ClearML results page: https://app.clear.ml/projects/323c182596f34d73934d0b7905eeb81d/experiments/42967e99b8f946f6a650972ca1b950a2/output/log
ClearML Monitor: GPU monitoring failed getting GPU reading, switching off GPU monitoring
2022-08-26 12:46:41,857 - clearml.Task - INFO - Waiting to finish uploads
2022-08-26 12:46:42,242 - clearml.Task - INFO - Completed model upload to https://files.clear.ml/tests/.pipelines/test%20pipeline%20monitor%20models/step1.42967e99b8f946f6a650972ca1b950a2/models/model1.xxx
2022-08-26 12:46:43,065 - clearml.Task - INFO - Finished uploading
None
after step1:
model1
Launching step [step2]
ClearML results page: https://app.clear.ml/projects/323c182596f34d73934d0b7905eeb81d/experiments/cee488fd878743c59badfae1a5b6216d/output/log
ClearML Monitor: GPU monitoring failed getting GPU reading, switching off GPU monitoring
2022-08-26 12:47:18,642 - clearml.Task - INFO - Completed model upload to https://files.clear.ml/tests/.pipelines/test%20pipeline%20monitor%20models/step2.cee488fd878743c59badfae1a5b6216d/models/model2.xxx
2022-08-26 12:47:18,946 - clearml.Task - INFO - Waiting to finish uploads
2022-08-26 12:47:19,352 - clearml.Task - INFO - Finished uploading
None
after step2:
model2    <--------------------------------------------------------------------- model1 missing
2022-08-26 12:47:34,996 - clearml.Task - INFO - Waiting to finish uploads
2022-08-26 12:47:35,522 - clearml.Task - INFO - Finished uploading
```
As noted by the arrow above, after step2, both model1 and model2 should be present on the pipeline task, but model1 is missing.

## Environment
* app.clear.ml
* ClearML SDK Version: 1.6.4
* Python Version: 3.9.12
* MacOS 12.5

**Comment by pollfly** [comment]
Hey @tonyd, thanks for the PR! Just letting you know that this issue has been resolved in [v1.7.0](https://github.com/allegroai/clearml/releases/tag/v1.7.0). Let us know if there are any issues :)

**Comment by jkhenning** [comment]
Closing this issue. Please reopen if it's still relevant.
